### PR TITLE
Avoid sharing state between tests

### PR DESCRIPTION
**Pull request by thewilkybarkid**
Relates to https://github.com/libero/content-api-bundle/pull/5#discussion_r228835169: one test only passed due to ordering. This removes the local attempt at sharing a kernel between test cases.

### CODE REVIEW

File: phpunit.xml.dist [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
     xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/7.2/phpunit.xsd" colors="true"
-    bootstrap="tests/bootstrap.php">
+    bootstrap="tests/bootstrap.php" executionOrder="random" resolveDependencies="true">
```
Comment (giorgiosironi) [inline phpunit.xml.dist:5]: The execution order should never matter unless there is unwanted interaction between tests (so it can be random)

Comment (thewilkybarkid) [reply phpunit.xml.dist:5]: This was found by trying out [Infection](https://infection.github.io/). (I wasn't aware PHPUnit 7.2 had added this as an option.)

File: tests/Functional/FunctionalTestCase.php [hunk 2]
```diff
@@ -4,84 +4,71 @@
...
 use Symfony\Component\HttpKernel\KernelInterface;
 use tests\Libero\ContentApiBundle\Functional\App\Kernel;
 use function tests\Libero\ContentApiBundle\capture_output;
 
-abstract class FunctionalTestCase extends TestCase
+abstract class FunctionalTestCase extends KernelTestCase
 {
-    /** @var Filesystem */
-    private static $filesystem;
+    /** @var ContainerInterface */
```
Comment (giorgiosironi) [inline tests/Functional/FunctionalTestCase.php:21]: is worth of a comment on why we are not sharing a `Kernel` between tests, but we share a `Container` (likely performance)

File: tests/Functional/FunctionalTestCase.php [hunk 3]
```diff
@@ -4,84 +4,71 @@
...
-                return $kernel->handle($request);
+            function () use ($request) : Response {
+                return self::$kernel->handle($request);
             },
             $content
         );
     }
 
-    private static function getTestCases() : iterable
+    final protected static function bootKernel(array $options = []) : KernelInterface
```
Comment (giorgiosironi) [inline tests/Functional/FunctionalTestCase.php:34]: It doesn't seem to strictly need to be a static method apart from inheriting the same interface as the Symfony-provided version. `static::` can be a way to signal that there is indeed something process-wide going on (storing some state on `self`).

**Review by giorgiosironi — APPROVED** [review]

File: phpunit.xml.dist (hunk 1 above)
Comment (stephenwf) [reply phpunit.xml.dist:5]: How did you find using Infection?

File: tests/Functional/FunctionalTestCase.php (hunk 2 above)
Comment (thewilkybarkid) [reply tests/Functional/FunctionalTestCase.php:21]: The container isn't shared.

File: tests/Functional/FunctionalTestCase.php (hunk 3 above)
Comment (thewilkybarkid) [reply tests/Functional/FunctionalTestCase.php:34]: A bunch of this is just for compatibility with Symfony < 4.1, will add a comment.

File: tests/Functional/FunctionalTestCase.php (hunk 2 above)
Comment (thewilkybarkid) [reply tests/Functional/FunctionalTestCase.php:21]: (Assuming `bootKernel` is called again. Symfony has `static::$container = null` in `ensureKernelShutdown()`, haven't added it here to the typehint can remain not-nullable. It does call `reset()` so the container itself is different.)

File: phpunit.xml.dist (hunk 1 above)
Comment (thewilkybarkid) [reply phpunit.xml.dist:5]: Seems very promising. Always been interesting in the idea but never got around to trying it out. It finds some false positives, but does find some cases where there's no coverage.

**Pull request merged**